app/app.go

```diff
@@ -103,6 +104,113 @@ func (app *App) GetSortedSolutionEnabledNotes() (allNoteIDs []string) {
+// changes TuneForNotes and NoteApplyOrder
+func (app *App) removeFromConfig(noteID string) {
+	i := sort.SearchStrings(app.TuneForNotes, noteID)
+	if i < len(app.TuneForNotes) && app.TuneForNotes[i] == noteID {
+		// remove noteID from the configuration 'TuneForNotes'
+		app.TuneForNotes = append(app.TuneForNotes[0:i], app.TuneForNotes[i+1:]...)
+	}
+	i = app.PositionInNoteApplyOrder(noteID)
+	if i < 0 {
+		system.WarningLog("noteID '%s' not found in configuration 'NoteApplyOrder'", noteID)
+	} else if i < len(app.NoteApplyOrder) && app.NoteApplyOrder[i] == noteID {
+		// remove noteID from the configuration 'NoteApplyOrder'
+		app.NoteApplyOrder = append(app.NoteApplyOrder[0:i], app.NoteApplyOrder[i+1:]...)
+	}
+}
+
+// IsNoteApplied checks, if a note is applied or not
+// return true, if note is already applied
+// return false, if not is NOT applied
+func (app *App) IsNoteApplied(noteID string) (string, bool) {
+	rval := ""
+	ret := false
+	// ANGI TODO - is check against state file still ok or should
+	// the check switched to first check NOTE_APPLY_ORDER and later
+	// the state file
+	sfile, err := os.Stat(app.State.GetPathToNote(noteID))
+	if err == nil {
+		// state file for note already exists
+		// check, if note is part of NOTE_APPLY_ORDER
+		if app.PositionInNoteApplyOrder(noteID) < 0 { // noteID not yet available
+			// bsc#1167618
+			// check, if state file is empty - seems to be a
+			// left-over of the update from saptune V1 to V2
+			if sfile.Size() == 0 {
+				// remove old, left-over state file and go
+				// forward to apply the note
+				os.Remove(app.State.GetPathToNote(noteID))
+			} else {
+				// data mismatch, do not apply the note
+				system.WarningLog("note '%s' is not listed in 'NOTE_APPLY_ORDER', but a non-empty state file exists. To prevent configuration mismatch, please revert note '%s' first and try again.", noteID, noteID)
+				rval = "mismatch"
+				ret = true
+			}
+		} else { // note applied
+			ret = true
+		}
+	}
+	return rval, ret
+}
+
+// NoteSanityCheck checks, if for all notes listed in
+// NoteApplyOrder and TuneForNotes a note definition file exists.
+// if not, remove the NoteID from the variables, save the new config and
+// inform the user
+func (app *App) NoteSanityCheck() error {
+	// app.NoteApplyOrder, app.TuneForNotes
+	errs := make([]error, 0, 0)
+	for _, note := range app.NoteApplyOrder {
+		if _, exists := app.AllNotes[note]; exists {
+			// note definition file for NoteID exists
+			continue
+		}
+		// bsc#1149205
+		// noteID available in apply order list, but no note definition
+		// file found. May be removed or renamed.
+		system.ErrorLog("the Note ID '%s' is not recognised by saptune, but is listed in the apply order list.\nMay be you removed or renamed the associated Note definition file on command line without reverting the Note first.\nWe will now remove the NoteID from the apply order list to prevent further confusion.", note)
```

Comment:
```suggestion
		system.ErrorLog("The Note ID '%s' is not recognized by saptune, but it is listed in the apply order list.\nMay be the associated Note definition file was removed or renamed via command line without previously reverting the Note.\nSaptune will now remove the NoteID from the apply order list to prevent further confusion.", note)
```

Reply:
many thanks for rephrasing the error message. I highly appreciate it.